libraries/AP_HAL_Linux/SPIDevice.cpp

```diff
@@ -251,11 +258,47 @@ bool SPIDevice::set_speed(AP_HAL::Device::Speed speed)
 bool SPIDevice::transfer(const uint8_t *send, uint32_t send_len,
                          uint8_t *recv, uint32_t recv_len)
 {
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_ALTERA
+    struct spi_ioc_transfer msgs[1] = { };
+#else
     struct spi_ioc_transfer msgs[2] = { };
     unsigned nmsgs = 0;
+#endif
 
     assert(_bus.fd >= 0);
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_ALTERA
+    uint8_t len_t = 0;
+    uint8_t i     = 0;
+
+    uint8_t send_t[32] = { 0, };
+    uint8_t recv_t[32] = { 0, };
```

Comment:
what's the problem with the current transfer implementation? Your approach won't work if you have a transfer greater than 32 bytes.  I guess this implementation is here only because you rebased your branch and you were using an older implementation.  The current implementation is working on several different boards with different kernel versions. Could you give it a try rather than adding more ifdefs? Or give some details on why it doesn't work for you.